relay-ourlogs/src/vercel_to_sentry.rs

```diff
@@ -159,20 +159,28 @@ fn map_vercel_level_to_sentry(level: VercelLogLevel) -> OurLogLevel {
     }
 }
 
-fn get_trace_id(trace_id: Option<String>) -> Annotated<TraceId> {
+fn get_trace_id(trace_id: Option<String>, request_id: Option<String>) -> Annotated<TraceId> {
     match trace_id {
-        Some(s) if !s.is_empty() => match TraceId::from_str(&s) {
+        Some(s) if !s.is_empty() => match s.parse::<TraceId>() {
             Ok(id) => Annotated::new(id),
-            Err(_) => {
-                let mut meta = Meta::default();
+            Err(_) => Annotated::new_with_meta(TraceId::random(), |meta| {
                 meta.add_remark(Remark::new(RemarkType::Substituted, "trace_id.invalid"));
-                Annotated(Some(TraceId::random()), meta)
-            }
+            }),
         },
         _ => {
-            let mut meta = Meta::default();
-            meta.add_remark(Remark::new(RemarkType::Substituted, "trace_id.missing"));
-            Annotated(Some(TraceId::random()), meta)
+            if let Some(request_id_string) = request_id {
+                if let Ok(id) = request_id_string.parse::<TraceId>() {
+                    Annotated::new_with_meta(id, |meta| {
+                        meta.add_remark(Remark::new(
+                            RemarkType::Substituted,
+                            "trace_id.replaced_with_request_id",
+                        ));
```

Comment:
Generally we don't do these remarks when creating data, because a user didn't explicitly send in our format and then we tinkered with it, it's just the logic we use to create our item.

E.g. you wouldn't do that in an SDK.

If this actually helps you debug, then it's fine to keep in.